test/test_nn.py

```diff
@@ -10511,6 +10511,15 @@ def test(nonlinearity, *args, **kwargs):
         test('threshold', 3, 2)
         test('threshold', 3, 2, inplace=True)
 
+    def test_pooling_shape(self, device):
+        def check(expected_out_shape, sizes, *args, **kwargs):
+            t = torch.randn(sizes, device=device)
+            self.assertEqual(torch.nn.functional.max_pool3d(t, *args, **kwargs).shape, expected_out_shape)
```

Comment:
The issue was for max_pool2d but you're only testing max_pool3d.

Reply:
max_pool2d for quantized tensors calls max_pool3d. Ideally this should test all pooling code, but since they all call the same function to compute output shape, 3D gives me more dimensions to test at once.